hazelcast/src/test/java/com/hazelcast/config/XMLConfigPreProcessorTest.java

```diff
@@ -500,6 +501,54 @@ public void testXmlDeniesDuplicateMemberAttributesConfig() throws Exception {
         configBuilder.build();
     }
 
+    @Test
+    public void testXmlVariableReplacementAsSubstring() throws Exception {
+
+        String xml = "<hazelcast>\n" +
+                "    <properties>\n" +
+                "        <property name=\"${env}-with-suffix\">local-with-suffix</property>\n" +
+                "        <property name=\"with-prefix-${env}\">with-prefix-local</property>\n" +
+                "    </properties>\n" +
+                "</hazelcast>";
+
+        System.setProperty("env", "local");
+
+        ByteArrayInputStream bis = new ByteArrayInputStream(xml.getBytes());
+        XmlConfigBuilder configBuilder = new XmlConfigBuilder(bis);
+        Config config = configBuilder.build();
+        Properties properties = config.getProperties();
+
+        assertEquals(properties.getProperty("local-with-suffix"), "local-with-suffix");
+        assertEquals(properties.getProperty("with-prefix-local"), "with-prefix-local");
+    }
+
+    @Test
+    public void testXmlImportWithVariableReplacementAsSubstring() throws Exception {
+        File file = createConfigFile("foo", "bar");
+        FileOutputStream os = new FileOutputStream(file);
+        String networkConfig = "<hazelcast>" +
+                "    <network>\n" +
+                "        <join>\n" +
+                "            <multicast enabled=\"false\"/>\n" +
+                "            <tcp-ip enabled=\"true\"/>\n" +
+                "        </join>\n" +
+                "    </network>\n" +
+                "</hazelcast>";
+        writeStringToStreamAndClose(os, networkConfig);
+
+        String xml = "<hazelcast>\n" +
+                "    <import resource=\"file://" + "${file}" + "\"/>\n" +
+                "</hazelcast>";
+
+        ByteArrayInputStream bis = new ByteArrayInputStream(xml.getBytes());
+        System.setProperty("file", file.getAbsolutePath());
+        XmlConfigBuilder configBuilder = new XmlConfigBuilder(bis);
+        Config config = configBuilder.build();
+        JoinConfig join = config.getNetworkConfig().getJoin();
+        assertFalse(join.getMulticastConfig().isEnabled());
```

Comment:
Personally I would not use the network override to detect if a file is correctly imported. I needed to look at the config to determine what the defaults were. Personally I would go for adding a property in the property section. 

```
<properties>
   <property name="foo">bar</name>
</properties>
```

And then check if the property 'foo' has value 'bar'.